app/containers/Itunes/index.js

```diff
@@ -3,42 +3,67 @@
  * Itunes
  *
  */
-
 import React, { useState } from 'react';
 // import PropTypes from 'prop-types'
 import { connect } from 'react-redux';
 import { injectIntl } from 'react-intl';
 // import { Helmet } from 'react-helmet';
 import debounce from 'lodash/debounce';
+import get from 'lodash/get';
+import { injectSaga } from 'redux-injectors';
 
 import { createStructuredSelector } from 'reselect';
 import { compose } from 'redux';
-// import { useInjectSaga } from '@utils/injectSaga'
-import makeSelectItunes from './selectors';
-// import saga from './saga';
 
-export function Itunes() {
+import {makeSelectItunes, selectItunesData, selectItunesError, selectItunesName} from './selectors'
+import { isEmpty } from 'lodash';
+import { itunesCreators } from './reducer';
+
+import itunesSaga from './saga';
+
+export function Itunes({ itunesName, itunesData={}, ituneError, dispatchItunesList, dispatchClearItunesList }) {
   const [name, SetName] = useState();
   const [data, SetData] = useState('');
   const API = 'https://itunes.apple.com/search?term=';
-  // useInjectSaga({ key: 'itunes', saga })
 
   const handleOnChange = (txt) => {
     SetName(txt);
   };
   const handleOnClick = (txt) => {
-    handleApi(txt);
-  };
-  const handleApi = (txt) => {
-    fetch(API + txt)
-      .then((res) => res.json())
-      .then((data) => SetData(data.results));
+    if (!isEmpty(txt)) {
+      dispatchItunesList(txt);
+      console.log(get(itunesData,'results'));
+    } else {
+      dispatchClearItunesList();
```

Comment:
shift this logic to `handleOnChange`, when the user deletes the search term it should automatically clear the list.

Reply:
if you have debounce implemented you dont need the click event simply move this code block to `handleOnChange`